src/jm_lzh.h

```diff
@@ -46,12 +46,12 @@ extern void (* LZH_DecompressDisplayVector)(
 bool LZH_Startup();
 void LZH_Shutdown();
 
-int LZH_Compress(
+int32_t LZH_Compress(
```

Comment:
I'm not sure I understand. I had to change them from int to int32_t because the compiler was complaining that it was expecting int32_t, I wish I had the exact error message ready to hand.

Reply:
"codesize" is unsigned. If you cast it before returning, like I did above, the compiler will not complain (I hope).

Reply:
It could have been because of https://github.com/bibendovsky/bstone/blob/develop/src/3d_main.cpp#L8455
and
https://github.com/bibendovsky/bstone/blob/develop/src/3d_main.cpp#L8482

I can try to revert the declarations to their original versions and change the return part as you suggested.

Reply:
Also compare

https://github.com/bibendovsky/bstone/blob/develop/src/jm_lzh.h#L54

with

https://github.com/bibendovsky/bstone/blob/develop/src/jm_lzh.cpp#L993

I can't find a way around this (to get it to compile without error), even with the recasting bit inside the function.

```
~/bstone-vita/src/jm_lzh.cpp: In function 'int32_t LZH_Decompress(const void*, void*, uint32_t, uint32_t)':
~/bstone-vita/src/jm_lzh.cpp:993:9: error: ambiguating new declaration of 'int32_t LZH_Decompress(const void*, void*, uint32_t, uint32_t)'
 int32_t LZH_Decompress(
         ^~~~~~~~~~~~~~
In file included from ~/bstone-vita/src/jm_lzh.cpp:53:0:
~/bstone-vita/src/jm_lzh.h:54:5: note: old declaration 'int LZH_Decompress(const void*, void*, uint32_t, uint32_t)'
 int LZH_Decompress(
     ^~~~~~~~~~~~~~
```

Reply:
I think LZH_Decompress is meant to be int32_t (cf. https://github.com/bibendovsky/bstone/blob/develop/src/jm_lzh.cpp#L1000 ). I now see your point about LZH_Compress. Commit 8a713d2 reflects this position. I have to admit that my understanding of this part of the code is very limited, but this is the best I can come up with at the moment.

Reply:
... and commit 67395d1 compiles with LZH_Decompress as "int"

Reply:
FWIW, both LZH_Compress and LZH_Decompress were of type "long" in both declaration and definition in the original release:
https://github.com/johndrinkwater/blake-stone/blob/master/JM_LZH.H#L19
https://github.com/johndrinkwater/blake-stone/blob/master/JM_LZH.C#L1049